text/0000-lang-team-mcp.md

```diff
@@ -0,0 +1,216 @@
+
+## Automation
+
+We will have a bot that monitors proposals. The bot will take the following automated actions (though the details here are subject to change as we experiment).
+
+* When an issue is opened with the Major Change Proposal template, it will be tagged as a "draft proposal".
+    * An automated bot will create a dedicated Zulip topic in the `#t-lang/major changes` stream (same as the compile team process).
+* (Optional) After N days without a comment from a lang-team member, an automated bot will indicate that the issue will be closed soon due to inactivity. After N more days, the issue is autoclosed.
+
+## Transparency around bandwidth
+
+As part of this process, the lang-team plans to organize and expose more clearly which project groups each member is currently involved with (likely using a Github project to expose details and other information). Similarly, we plan to communicate more clearly the sorts of proposals we are looking for at any period of time via blog posts or information on the [lang-team website].
+
+[lang-team website]: https://lang-team.rust-lang.org/
+
+## Transitioning to the new system
+
+There are a number of existing RFCs, many of which have received quite a lot of work, that have to be transitioned to the new system. We do not want to just "auto close" all of those RFCs because in many cases they represent a significant amount of effort and "near consensus".
+
+Our plan is to first create the new system and encourage folks to create proposals who wish to. Presuming our experience with the new system is positive, and once we have "ironed out the kinks", we will begin a process to port over the existing RFCs. 
+
+The precise plans will be determined then, but they will likely include:
+
+* A review of the RFCs to try and find those lang-team liaisons where possible.
+* Announcing a date when we will complete the migate and posting a notice on existing RFCs. This notice would encourage others to try out the new process. After that date expires, existing RFCs would be closed.
+* An alternative might be to "auto-migrate" all existing proposals via automation to create proposals, perhaps spaced out in groups.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* There is an additional step to prepare an RFC, which indicates more overhead and procedures to be followed.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+* We could have proposals be opened on the RFC repository
+    * Why? Central place for all proposals across teams.
+    * Why not? The RFC repository is no longer a "high signal" place to watch for "changes that have traction".
+* We could try this for all teams at once.
+    * Why? Uniform process is good.
+    * Why not? Not sure how it will work, and each team has somewhat different needs.
+* We could have MCPs be opened as a PR
```

Comment:
IMO, the main drawback of using issues and not PRs is that it is hard to comment on a particular line of the text, as I'm doing here, and thread conversations off that.

A possible counterpoint being that we expect issues to be high-level and short, not requiring highly threaded discussions. I'm not sure that's always going to be the case, though.

Reply:
So, one question is: where do we expect the discussion about a proposal to happen -- and, perhaps, who should be the participants?

In particular, on the compile team anyway, we've been pushing that discussion to Zulip, and I think that's been a net win over using github issues. The github issue is reserved for procedural or status updates.

We also don't generally have "long-lived" MCPs -- if there is a proposal and it changes drastically, we close it and open a new one.

Reply:
But to reply to your original point, I also generally find it quite hard to follow PRs discussions, with their combination of "in-line comments" and "main-line comments" and so forth. There isn't really a perfect setup here.

Reply:
In terms of "who should be the participants", your comments teased out something. I am thinking that perhaps the idea of the MCP should be more about seeking feedback **from the lang-team** (or perhaps other intersecting teams, key stakeholders, etc) than feedback **from the wider community**. I imagine the "wider community" feedback probably happening earlier, on internals or elsewhere, and it would be summarized as part of the MCP.

Reply:
(That said, I will say that I'm a bit nervous about lang-team discussions moving too much to Zulip, because I fear it will stress the moderation story there.)